### PR TITLE
Drag drop move component

### PR DESCRIPTION
**Pull request by eireland**
Drag and drop component tiles in free tile layout.
Click on title text brings up edit title mode. Need to fix mousedown and drag allows for both edit and drag of tile.

### CODE REVIEW

**Review by kswenson — CHANGES_REQUESTED** [review]
👍 Generally looks good -- the usual handful of suggestions on various topics. There's obviously some z-index issues to work out with some of the elements but those can wait for a subsequent PR.

**Comment by kswenson** [comment]
Also, it seems like there must be a simplification of the title bar code lurking somewhere nearby, but I can't quite wrap my head around it and in any case it can wait for a subsequent PR.

**Review by bfinzer — APPROVED** [review]
👍🏻Deferring to Kirk

**Review by kswenson — APPROVED** [review]
👍 Changes look good

File: v3/src/components/dnd-detect-collision.ts [hunk 1]
```diff
@@ -21,6 +21,13 @@ export const dndDetectCollision: CollisionDetection = (args) => {
   // rectIntersection (for keyboard sensor)
   const collisions = args.pointerCoordinates ? pointerWithin(args) : rectIntersection(args)
 
+  // if this is a tile drag, then ignore all collisions other than the container
+  if (args.active.data.current?.type === "tile") {
+    const containers = args.droppableContainers.filter(({id}) => id === "codap-container-drop")
+    // apply the collection detection function specified by the tile
```
Comment (kswenson) [inline v3/src/components/dnd-detect-collision.ts:27]:
```suggestion
```
This comment is not relevant in this context.

**Review by kswenson — APPROVED** [review]
👍 Looks good -- just one stale comment to remove

**Pull request merged**